djangosaml2/views.py

```diff
@@ -177,17 +175,20 @@ def login(request,
         return HttpResponseRedirect(get_location(result))
     elif binding == BINDING_HTTP_POST:
         if not post_binding_form_template:
+            # use the html provided by pysaml2
             return HttpResponse(result['data'])
-        try:
-            params = get_hidden_form_inputs(result['data'][3])
+        else:
+            # manually get request XML to build our own template
+            request_id, request_xml = client.create_authn_request(
```

Comment:
I don't like doing this :-(.
We already have a `result` from `prepare_for_authenticate`, now we should disregard it and ask for a new, just because we can "parse it better"?
I take it that you're trying to fine-tune the code and I appreciate it, but this is ain't broke (it works!) and I prefer stability here. There's really no need to do this in my opinion.